js/modules/modules.go

```diff
@@ -69,31 +65,108 @@ type HasModuleInstancePerVU interface {
 	NewModuleInstancePerVU() interface{}
 }
 
+// IsModuleV2 ... TODO better name
+type IsModuleV2 interface {
+	NewModuleInstance(InstanceCore) Instance
```

Comment:
Does this make `NewModuleInstancePerVU()` obsolete? Essentially every V2 module will return a new instance per VU?

Reply:
yes, but we still use it, and somebody else, might too, so I prefer to deprecate it at a later point and remove at an even later one